lamindb/models/artifact.py

```diff
@@ -1863,10 +1890,21 @@ def from_dataframe(
                 )
                 return artifact
 
-            curator = DataFrameCurator(artifact, schema)
-            curator.validate()
-            artifact.schema = schema
-            artifact._curator = curator
+            if schema.itype == "Composite" and features is not None:
+                if schema is not None and schema.itype == "Composite":
+                    validation_schema = next(iter(schema.slots.values()))
+                    temp_df = pd.DataFrame([features])
+                    external_curator = DataFrameCurator(temp_df, validation_schema)
+                    external_curator.validate()
+
+                artifact._external_features = features
+            else:
+                # Regular schema means DataFrame validation
```

Comment:
Need to pay attention to the edge case where `.attrs` needs to be validated in addition to an external slot.